unittesting/core/st3/fast_runner.py

```diff
@@ -0,0 +1,167 @@
+            start_time=None
+        ):
+            if start_time is None:
+                start_time = time.time()
+
+            try:
+                send_value = condition()
+            except Exception as e:
+                _continue_testing(deferred, throw_value=e)
+                return
+
+            if send_value:
+                _continue_testing(deferred, send_value=send_value)
+            elif (time.time() - start_time) * 1000 >= timeout:
+                error = TimeoutError(
+                    'Condition not fulfilled within {:.2f} seconds'
+                    .format(timeout / 1000)
+                )
+                _continue_testing(deferred, throw_value=error)
+            else:
+                defer(period, _wait_condition, deferred, condition, period, timeout, start_time)
+
+        def _handle_error(e):
+            stopTestRun = getattr(result, 'stopTestRun', None)
+            if stopTestRun is not None:
+                stopTestRun()
+            self.finished = True
+            raise e
+
+        def _stop_testing():
+            with warnings.catch_warnings():
+                stopTestRun = getattr(result, 'stopTestRun', None)
+                if stopTestRun is not None:
+                    stopTestRun()
+
+            stopTime = time.time()
+            timeTaken = stopTime - startTime
+            result.printErrors()
+            if hasattr(result, 'separator2'):
+                self.stream.writeln(result.separator2)
+            run = result.testsRun
+            self.stream.writeln("Ran %d test%s in %.3fs" %
+                                (run, run != 1 and "s" or "", timeTaken))
+            self.stream.writeln()
+
+            expectedFails = unexpectedSuccesses = skipped = 0
+            try:
+                results = map(len, (result.expectedFailures,
+                                    result.unexpectedSuccesses,
+                                    result.skipped))
+            except AttributeError:
+                pass
+            else:
+                expectedFails, unexpectedSuccesses, skipped = results
+
+            infos = []
+            if not result.wasSuccessful():
+                self.stream.write("FAILED")
+                failed, errored = len(result.failures), len(result.errors)
+                if failed:
+                    infos.append("failures=%d" % failed)
+                if errored:
+                    infos.append("errors=%d" % errored)
+            else:
+                self.stream.write("OK")
+            if skipped:
+                infos.append("skipped=%d" % skipped)
+            if expectedFails:
+                infos.append("expected failures=%d" % expectedFails)
+            if unexpectedSuccesses:
+                infos.append("unexpected successes=%d" % unexpectedSuccesses)
+            if infos:
+                self.stream.writeln(" (%s)" % (", ".join(infos),))
+            else:
+                self.stream.write("\n")
+
+        _start_testing()
```

Comment:
If we could replace the function calls above by `defer(....)`, it is strong reason that we should use the setting `yield_timeout`.